### ISSUE TITLE
editora / editoratype duplication with crossref

### ISSUE THREAD

**Issue by jackweinbender** [dcpurton/biblatex-sbl#46]
I've noticed that it you use the editora/editoratype combinations with crossrefs, the editor is printed twice.

![screen shot 2016-11-28 at 12 40 15 am](https://cloud.githubusercontent.com/assets/1544859/20658535/4acc43c8-b503-11e6-9109-e31b799fabbf.png)

```
% bug.tex
\documentclass{article}
\usepackage[backend=biber,style=sbl]{biblatex}
\addbibresource{min-bib.bib}

\begin{document}
\nocite{*}
\printbibliography[title=Bibliography]

\end{document}
```

```
%min-bib.bib
@incollection{sanders2001,
	Author = {Sanders, James A.},
	Crossref = {flint2001},
	Date-Added = {2016-07-05 21:11:38 +0000},
	Date-Modified = {2016-07-06 19:45:08 +0000},
	Pages = {7--26},
	Read = {1},
	Shorttitle = {Canon as Dialogue},
	Title = {Canon as Dialogue}}
@book{flint2001,
	Booktitle = {The Bible at Qumran: Text, Shape, and Interpretation},
	Call-Number = {BM487 .B52 2001},
	Cataloged = {1},
	Date-Added = {2016-07-05 21:11:38 +0000},
	Date-Modified = {2016-11-28 06:07:05 +0000},
	Editor = {Flint, Peter W.},
	Editora = {Kim, Tae Hun},
	Editoratype = {with the assistance of},
	Location = {Grand Rapids},
	Own = {1},
	Publisher = {Wm. B. Eerdmans},
	Series = {Studies in the Dead Sea and Related Literature},
	Shortseries = {SDSRL},
	Shorttitle = {The Bible at Qumran},
	Title = {The Bible at Qumran: Text, Shape, and Interpretation},
	Year = {2001}}
```

**Comment by dcpurton** [comment]
This was already fixed in the dev version, however your report highlighted another problem which I've just also fixed and pushed to the dev version. Try now and see if it does what you want

However: Instead of using `editora` and `editoratype`, I think you should use `witheditor` and `witheditortype`. These are custom fields to `biblatex-sbl` though so you might want to avoid them. It gives a slightly different output that I *think* better matches the SBL guidelines. Consider the punctuation of *TLOT* on page 211 of the SBLHS. This isn't possible using `editora` and `editoratype`. If you use `witheditor` then `witheditor` and `witheditortype` will be separated from the `editor` with a comma rather than a period in the bibliography and list of abbreviations.

I also made the call to put the `witheditor` (and `withauthor` and `withtranslator`) at the start of the citation along with the primary `author`, `editor`, or `translator`. This is more controversial. There's no example like this given in the handbook and I have seen it done a number of ways in publications. In your particular example, it will mean `\cite{flint2001}` will produce: Peter W. Flint, ed., with the assistance of Tae Hun Kim, *The Bible at Qumran*…

Overall I prefer this solution because it ties the `witheditor` to the `editor` in both the citation and the bibliography. If you don't do this, and use `editora` instead, then you can get ambiguous output. Consider this example:

```
@book{test,
  author = {An Author},
  title = {Title},
  date = {2016},
  editor = {An Editor},
  editora = {Another Author},
  editoratype = {with}
}
```

I actually want `editora` associated with `author`, but there's no way for `biblatex` to know this. Instead it will get printed after `editor`.

I think I get more sensible output if instead I use:

```
@book{test,
  author = {An Author},
  title = {Title},
  date = {2016},
  editor = {An Editor},
  withauthor = {Another Author}
}
```

In this case `\cite{test}\par\cite{test}\printbibliography` will produce:

An Author, with Another Author, *Title,* ed. An Editor (2016).
Author, *Title.*
Author, An, with Another Author. *Title.* Edited by An Editor. 2016.

I am however open to discussion on this issue if others would like to chime in.

(Aside: Documentation in the dev version is out of date at the moment.)

**Comment by dcpurton** [comment]
Fixed in v0.8.1
